jib-core/src/main/java/com/google/cloud/tools/jib/frontend/CredentialRetrieverFactory.java

```diff
@@ -227,13 +223,18 @@ CredentialRetriever dockerConfig(
     };
   }
 
-  private Credential retrieveFromDockerCredentialHelper(
-      Path credentialHelper, DockerCredentialHelperFactory dockerCredentialHelperFactory)
+  @VisibleForTesting
+  void setDockerCredentialHelperFactory(
```

Comment:
I would lean towards having `dockerCredentialHelperFactory` be final and just passing it in via a testing constructor.